lib/fs.js

```diff
@@ -609,6 +609,12 @@ fs.read = function(fd, buffer, offset, length, position, callback) {
     };
   }
 
+  if (length === 0) {
+    return process.nextTick(function() {
```

Comment:
Actually, this calls the redefined `callback` function (that starts on line 600), so the right arguments should be passed to the callback function, namely: `null`, `''`, and `0`.

Unless I'm missing something.